src/commands/generate/generators/auth/action.js

```diff
@@ -1,9 +1,99 @@
+
 module.exports = async function action (params, utils, command) {
+  let { args } = params
+  let { writeFile, npmCommands } = utils
+  let { installAwsSdk } = npmCommands
+  let error = require('./errors')(params, utils)
   let project = params.inventory.inv._project
   let generate = require('../_generate')
 
-  let manifest = require('./manifest')
 
-  // Run the generic generator
-  generate({ manifest, command, project, utils })
+  let plugins = project.arc.plugins
+  if (plugins.includes('arc-plugin-oauth')) {
+    return error('oauth_plugin_already_installed')
+  }
+
+  let authType = args.t || args.type
+  if (authType === 'oauth') {
+    let manifest = require('./oauth-manifest')
+    generate({ manifest, command, project, utils })
+  }
+  else if (!authType || authType === 'magic-link') {
+
+    const prefsFile = project.localPreferencesFile
+    const { readFileSync } = require('fs')
+    let prefs = readFileSync(prefsFile, 'utf8')
+    if (!/@sandbox-startup/.test(prefs)) {
+      prefs += `@sandbox-startup
+node ./scripts/seed-users.js`
+      writeFile(prefsFile, prefs)
+    }
+
+    // Install Dependencies
+    installAwsSdk()
+
+    let manifest = require('./magic-manifest')
+
+    let { writeJsonSchema } = require('../scaffold/jsonschema')
+
+
+    const modelName = {
+      singular: 'user',
+      capSingular: 'User',
+      plural: 'users',
+      capPlural: 'Users'
+    }
+    const routeName = 'users'
+    const schema = {
```

Comment:
Might be better to move this stuff to a different file to make the action more readable.

Reply:
You are right. Done.